apps/wallet/src/ui/app/helpers/NftClient.ts

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import {
+    SuiObject, SuiMoveObject, is
+} from '@mysten/sui.js';
+
+import type {
+    GetObjectDataResponse,
+    JsonRpcProvider
+} from '@mysten/sui.js';
+
+export interface WithIds {
+    objectIds: string[];
+}
+
+
+type FetchFnParser<RpcResponse, DataModel> = (
+    typedData: RpcResponse,
+    suiObject: SuiObject,
+    rpcResponse: GetObjectDataResponse
+) => DataModel | undefined;
+
+type SuiObjectParser<RpcResponse, DataModel> = {
+    parser: FetchFnParser<RpcResponse, DataModel>;
+    regex: RegExp;
+};
+
+type ID = {
+    id: string;
+};
+
+type Bag = {
+    type: string;
+    fields: {
+        id: ID;
+        size: number;
+    };
+};
+
+type ArtNftRpcResponse = {
+    logical_owner: string;
+    bag: Bag;
+};
+
+type ArtNftRaw = {
+    id: string;
+    logicalOwner: string;
+    bagId: string;
+};
+
+type DomainRpcBase<T> = {
+    id: ID;
+    name: {
+        type: string;
+        fields: {
+            dummy_field: boolean;
+        };
+    };
+    value: {
+        type: string;
+        fields: T;
+    };
+};
+
+type UrlDomainRpcResponse = DomainRpcBase<{
+    url: string;
+}>;
+
+type DisplayDomainRpcResponse = DomainRpcBase<{
+    description: string;
+    name: string;
+}>;
+
+
+type NftDomains = {
+    url: string;
+    name: string;
+    description: string;
+};
+
+
+export type ArtNft = {
+    nft: ArtNftRaw
+    fields?: Partial<NftDomains>
+};
+
+const ArtNftRegex =
```

Comment:
Does it mean that I can publish a type `nft::NFT` and it will be treated in a special way?

Reply:
Actually, no. The protocol was changed several times, just a legacy of naming. Will change it.